guava/src/main/java/org/jdbi/v3/guava/GuavaCollectors.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.guava;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.stream.Collector;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.ImmutableSortedSet;
+
+import org.jdbi.v3.tweak.CollectorFactory;
+
+/**
+ * Provide Collector instances that create Guava collection types, especially immutable
+ * versions of the base Java collections.
+ */
+public class GuavaCollectors {
+    /**
+     * @return a collector into {@code ImmutableList<T>}
+     */
+    public static <T> Collector<T, ImmutableList.Builder<T>, ImmutableList<T>> immutableList() {
```

Comment:
We should match the `Collectors` method naming and prepend all the factory method names with `to`. This allows for fluent usage e.g. `.collect(toImmutableList())`